cedar-policy-core/src/parser.rs

```diff
@@ -1059,4 +1060,49 @@ mod parse_tests {
             }
         })
     }
+
+    #[test]
+    fn unexpected_token_errors() {
+        #[track_caller]
+        fn assert_labeled_span(policy: &str, label: impl Into<String>) {
+            assert_matches!(parse_policy(None, policy), Err(e) => {
+                let actual_label = e.labels().and_then(|l| {
+                    l.exactly_one()
+                        .ok()
+                        .expect("Assumed that there would be exactly one label if labels are present")
+                        .label()
+                        .map(|l| l.to_string())
+                });
+                assert_eq!(Some(label.into()), actual_label, "Did not see expected labeled span.");
+            });
+        }
+
+        // Don't list out all the special case identifiers
+        assert_labeled_span("@", "expected identifier");
+        assert_labeled_span(
+            "permit(principal, action, resource) when { principal.",
+            "expected identifier",
+        );
+
+        // We specifically want `when` or `unless`, but we previously listed all
+        // identifier tokens, so this is an improvement.
+        assert_labeled_span(
+            "permit(principal, action, resource)",
+            "expected `;` or identifier",
+        );
+        assert_labeled_span("@if(\"a\")", "expected `@` or identifier");
```

Comment:
this one should be `@`, `permit`, or `forbid`.  (doesn't have to be in this PR, but since we have comments on some of the other tests noting what error message would be even better)